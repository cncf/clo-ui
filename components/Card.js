module.exports = require('../dist/cjs/components/Card.js');
