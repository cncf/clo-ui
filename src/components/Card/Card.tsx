import classNames from 'classnames';
import { isUndefined } from 'lodash';
import React from 'react';

import styles from './Card.module.css';

export interface ICardProps {
  wrapperClassName?: string;
  className?: string;
  children: JSX.Element;
  hoverable: boolean;
  onClick?: () => void;
}

export const Card: React.FC<ICardProps> = (props: ICardProps) => {
  return (
    <div className={props.wrapperClassName} role="listitem">
      <div
        className={classNames(
          'card rounded-0 p-0 h-100 mw-100 d-flex text-reset text-decoration-none',
          props.className,
          {
            [styles.hoverable]: props.hoverable,
          }
        )}
        onClick={props.onClick || undefined}
        role={!isUndefined(props.onClick) ? 'button' : ''}
      >
        {props.children}
      </div>
    </div>
  );
};
