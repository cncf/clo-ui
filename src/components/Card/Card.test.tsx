import { render, screen } from '@testing-library/react';

import { Card } from './Card';

const defaultProps = {
  hoverable: false,
};

describe('Card', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(
      <Card {...defaultProps}>
        <>card content</>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(
      <Card {...defaultProps}>
        <>card content</>
      </Card>
    );

    expect(screen.getByText('card content')).toBeInTheDocument();
  });
});
