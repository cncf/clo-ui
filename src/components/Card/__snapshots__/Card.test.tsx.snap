// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Card creates snapshot 1`] = `
<DocumentFragment>
  <div
    role="listitem"
  >
    <div
      class="card rounded-0 p-0 h-100 mw-100 d-flex text-reset text-decoration-none"
      role=""
    >
      card content
    </div>
  </div>
</DocumentFragment>
`;
